lib/matplotlib/pyplot.py

```diff
@@ -688,6 +689,18 @@ def savefig(*args, **kwargs):
     return res
 
 
+
+def encode_as(**kwargs):
+    """ Equivalent of savefig, but does not store to a file, but returns a bytestring
```

Comment:
Fair enough, duplicate code is evil. I also copied code from `test_subplots.py` to `test_subplots_iterator.py`. Quite interested in your opinion about that.